### PR TITLE
fix(cl-staked): derive stakedLiquidityInRange from edges + clamp negatives

### LINKED ISSUE #719: Fix asymmetric stakedLiquidityInRange updates causing negative drift on CL pools
## What to build

The CL aggregator field `stakedLiquidityInRange` drifts negative on 8 flagship Optimism CL pools because counter and edge-net updates are asymmetric. Fix in two layers:

1. **Tactical clamp** — mirror PR #718's pattern: in `LiquidityPoolAggregator.ts`, clamp `stakedLiquidityInRange` at zero with a `[NEG_STAKED_LIQ_GUARD]` log line. Stops nonsensical negative state from contaminating snapshots and downstream USD valuations.
2. **Structural fix** — replace the cached running counter with derivation from existing edge state at swap/snapshot time:

   ```
   stakedLiquidityInRange =
     Σ stakedTickEdgeNets[i]   for all i where stakedTickEdges[i] <= currentTick
   ```

   Per-pool edge lists are sparse (typically < 50 entries) so the O(edges) recomputation is acceptable. Touches `CLStakedLiquidity.ts`, `GaugeSharedLogic.ts:104-110`, `NFPMCommonLogic.ts:170-176`, and their tests.

### Root cause

In all three places that update `stakedLiquidityInRange`, the counter is gated on `isPositionInRange(currentTick) && sqrtPriceX96 !== 0n && !edgesRejected`, but the edge nets (`stakedTickEdgeNets`) are updated unconditionally. Three concrete drift paths:

1. **Pre-Initialize deposits** — when `sqrtPriceX96 === 0n` at deposit, edge nets update but counter doesn't; the first Swap then walks edges from a wrong baseline.
2. **Edge-merge rejection** — when `applyStakedPositionToEdges` rejects, neither field updates, but the sparse edge map may already disagree with prior deposits.
3. **NFPM Increase/DecreaseLiquidity between gauge deposit and withdraw** — gauge withdraw subtracts the position's *current* liquidity, not the deposited amount; if NFPM adjusted in between, the subtraction doesn't match the addition.

### Scope

The 8 OP pools below are the visible (negative-drift) cases. Pools with *positive* drift exist across all chains but are undetectable from outside ground truth. The fix applies indexer-wide; a full reindex from scratch with the structural derivation in place heals every affected pool.

| Pool | Name | `stakedLiquidityInRange` |
|---|---|---|
| `10-0x3241738149B24C9164dA14Fa2040159FFC6Dd237` | CL-100 USDC/WETH | -1,021,293,404,683,997 |
| `10-0x4742B2760837C4b080D675c6770437CDb703c877` | CL-1 wstETH/WETH | -121,252,338,856,633,149,252 |
| `10-0x516D6c9A965292209E29b342a0534c460A4de766` | CL-1 USDC/USDT | -341,090,808,107,851 |
| `10-0x93e797aeD72bD99d39B1283ecb2FCF3A266a871A` | CL-100 WETH/LUSD | -14,460,165,144,096,204,916,709 |
| `10-0x94c0A04C0d74571aa9EE25Dd6c29E2A36f5699aE` | CL-1 USDC/sDAI | -67,836,763,217,978,517,657 |
| `10-0xB7Dc5834458CF797Dc290378902D599119B7f02d` | CL-100 WETH/DAI | -8,360,575,769,757,300,668 |
| `10-0xF5Ad4f36664cF4D47e3Ee57E4f32b671436CD325` | CL-1 USDC/LUSD | -6,530,912,409,801,676,142,868 |
| `10-0xFfF03Dba4f0591327b1572c1D66c375ECB0cDfd8` | CL-200 wstETH/OP | -226,179,900 |

### Related context

- PR #684 fixed `stakedReserve0/1` (per-segment Uniswap v3 attribution) but did NOT touch `stakedLiquidityInRange`'s running-counter logic.
- PR #713 fixed `liquidityInRange` (the unstaked counter) for in-range Mint/Burn — different field.
- PR #718 clamps reserve0/1 negatives — this issue applies the same shape of clamp but for `stakedLiquidityInRange`.

## Acceptance criteria

- [ ] Tactical clamp lands: `stakedLiquidityInRange` cannot be persisted < 0 by the aggregator
- [ ] `[NEG_STAKED_LIQ_GUARD]` log line emitted when a clamp would trigger, capturing pool address, chain, and the would-have-been value
- [ ] Structural derivation replaces the running-counter writes in `GaugeSharedLogic.ts`, `NFPMCommonLogic.ts`, `CLStakedLiquidity.ts`
- [ ] Unit tests cover all three drift paths above (pre-Initialize deposit, edge-merge rejection, NFPM-mediated liquidity changes between gauge deposit and withdraw)
- [ ] Edge-sanity invariant test: after deposit + N swaps + withdraw, `stakedLiquidityInRange == 0`
- [ ] After full reindex on Optimism, none of the 8 pools above show negative `stakedLiquidityInRange`
- [ ] Type-check + `pnpm test` + `pnpm qa --write` clean

## Blocked by

None - can start immediately

### PR DESCRIPTION
**Pull request by FonsSicca**
## Summary

Closes #719. Two-layer fix for the asymmetric `stakedLiquidityInRange` updates that drove 8 OP CL pools to negative values:

1. **Structural derivation** — replaces the cached running counter at the three writer sites (`GaugeSharedLogic`, `NFPMCommonLogic`, `processTickCrossingsForStaked`) with a new `deriveStakedLiquidityInRange(currentTick, edges, nets)`. The counter and edge nets were asymmetrically gated on `sqrtPriceX96 !== 0n`, `!edgesRejected`, and `isPositionInRange`; the counter drifted on three concrete paths (pre-Initialize deposit, edge-merge rejection, out-of-range NFPM liquidity change between gauge stake and withdraw). Derivation pins counter ≡ edges on every write and self-heals pre-existing drift on the next touch.

2. **Tactical clamp** — adds a `[NEG_STAKED_LIQ_GUARD]` clamp at `updateLiquidityPoolAggregator` mirroring PR #718's reserve clamp, so a negative value can never persist even if a future writer bypasses derivation.

## Acceptance criteria (issue #719)
- [x] Tactical clamp lands — `stakedLiquidityInRange` cannot be persisted < 0
- [x] `[NEG_STAKED_LIQ_GUARD]` log line with `poolAddress`, `chainId`, `priorStakedLiqInRange`, `replacement`, `clampedTo`
- [x] Structural derivation replaces running-counter writes in `GaugeSharedLogic.ts`, `NFPMCommonLogic.ts`, `CLStakedLiquidity.ts`
- [x] Unit tests cover all three drift paths (pre-Initialize, edge-merge rejection, NFPM mid-stake)
- [x] Edge-sanity invariant test: Deposit + N tick moves + Withdraw → counter == 0n
- [x] `tsc --noEmit` + `pnpm test` + `pnpm qa --write` clean
- [ ] Full reindex on Optimism to clear the 8 currently-negative pools — deferred to ops; the fix self-heals on the next gauge/NFPM/swap event per pool

## Test plan
- [ ] CI green (`pnpm test`, lint, type-check)
- [ ] Spot-check one of the 8 affected OP pools (e.g. `10-0x3241738149B24C9164dA14Fa2040159FFC6Dd237`) after deploy: confirm `stakedLiquidityInRange` heals from negative toward consistency with `stakedTickEdges`/`stakedTickEdgeNets` after subsequent gauge/swap events
- [ ] Watch logs for `[NEG_STAKED_LIQ_GUARD]` — should be rare; any hit signals a remaining producer path that bypasses derivation
- [ ] Confirm no regression in `stakedReserve0/1` (per-segment attribution path is unchanged)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Fixed staked liquidity tracking inconsistencies across position operations (deposits, withdrawals, swaps)
  * Added safeguard to prevent negative staked liquidity values with warning logging

* **Tests**
  * Expanded test coverage for liquidity calculation edge cases and regressions

[Review Change Stack](https://app.coderabbit.ai/change-stack/velodrome-finance/indexer/pull/725)

### CODE REVIEW

**Review by WooSungD — APPROVED** [review]

**Comment by FonsSicca** [comment]
@coderabbitai review

File: src/Aggregators/Pool.ts [hunk 1]
```diff
@@ -298,6 +298,23 @@ export async function updateLiquidityPoolAggregator(
...
+  const stakedLiqReplacement =
+    diff.stakedLiquidityInRange ?? current.stakedLiquidityInRange ?? 0n;
+  const clampedStakedLiquidityInRange =
+    stakedLiqReplacement < 0n ? 0n : stakedLiqReplacement;
+  if (stakedLiqReplacement < 0n) {
+    context.log.warn(
+      `[NEG_STAKED_LIQ_GUARD][updateLiquidityPoolAggregator] field=stakedLiquidityInRange poolAddress=${current.poolAddress} chainId=${current.chainId} priorStakedLiqInRange=${current.stakedLiquidityInRange} replacement=${stakedLiqReplacement} clampedTo=${clampedStakedLiquidityInRange}`,
+    );
+  }
+
```
[the remark this answers is not in the document]
Comment (FonsSicca) [reply src/Aggregators/Pool.ts:320]: Addressed in f751b90: cleared diff.stakedLiquidityInRange in both parallel-array guard branches so the counter falls back to the prior consistent value when the edge pair is dropped.

File: test/Aggregators/CLStakedLiquidity.test.ts [hunk 2]
```diff
@@ -926,4 +940,114 @@ describe("CLStakedLiquidity", () => {
...
+      const target = 150n;
+
+      const noDbContext = {
+        log: { error: vi.fn(), warn: vi.fn(), info: vi.fn(), debug: vi.fn() },
+      } as unknown as handlerContext;
+
+      const walked = processTickCrossingsForStaked(
+        8453,
+        "0xPool",
+        -500n,
```
[the remark this answers is not in the document]
Comment (FonsSicca) [reply test/Aggregators/CLStakedLiquidity.test.ts:1036]: Addressed in f751b90: replaced "0xPool" with the existing CHAIN_ID / POOL_ADDRESS constants from the file's top-level setup (POOL_ADDRESS is a checksummed 0x1234... mock).
